lib/stages/applyGains.cpp

```diff
@@ -656,6 +656,16 @@ applyGains::calculate_gain(double timestamp, uint32_t freq_ind, std::vector<cflo
 
     if (update_new == nullptr) {
         WARN("No gains update is as old as the currently processed frame.");
+
+        // Look up frequency ID and print more info about late frame
+        auto it = freq_map.begin();
+        for (; it != freq_map.end(); ++it) {
+            if (it->second == freq_ind)
+                break;
+        }
+        WARN("Frame frequency ID: {}\nFrame timestamp: {}\nUpdate queue: {}", it->first, timestamp,
+             gains_fifo);
```

Comment:
If for some reason the `freq_ind` isn't found this would attempt to dereference a non existent entry. For that reason I don't think it makes sense to have the WARN outside of the loop, so you might as well move it inside and then you get to use a structured binding range based for, and rejoice in the wonders of C++17. Something like:
```suggestion
        for (const auto& [fm_id, fm_ind] : freq_map) {
            if (fm_ind == freq_ind) {
                WARN(
                    "Frame frequency ID: {}\nFrame timestamp: {}\nUpdate queue: {}",
                    fm_id, timestamp, gains_fifo
                );
                break;
            }
        }

```